api/http/server.go

```diff
@@ -1118,30 +1126,22 @@ func calculateNumberOfChunks(contentLength int64, isEncrypted bool) int64 {
 // Recommended value is 4 times the io.Copy default buffer value which is 32kB.
 const getFileBufferSize = 4 * 32 * 1024
 
-// bufferedReadSeeker wraps bufio.Reader to expose Seek method
-// from the provied io.ReadSeeker in newBufferedReadSeeker.
-type bufferedReadSeeker struct {
-	r *bufio.Reader
-	s io.ReadSeeker
-}
+var defaultDownloaderMode = "langos"
 
-// newBufferedReadSeeker creates a new instance of bufferedReadSeeker,
-// out of io.ReadSeeker. Argument `size` is the size of the read buffer.
-func newBufferedReadSeeker(readSeeker io.ReadSeeker, size int) bufferedReadSeeker {
-	return bufferedReadSeeker{
-		r: bufio.NewReaderSize(readSeeker, size),
-		s: readSeeker,
+func newDownloader(s langos.Reader, r *http.Request) io.ReadSeeker {
+	mode := r.URL.Query().Get("_downloader")
```

Comment:
This is used for testing. We should remove this option before merging to master, or to find a different way to set it.

Reply:
agree, this is not necessary :+1: